File: legacy-algorithm-study/BOJ/[4673] 셀프넘버.js
function selfNum(a){
  let mok  = parseInt(a/10,10)
  let div = parseInt(a%10,10)
 
  a = mok+ div + a
  return a
}
let arr = new Array(10000)
arr.fill(false)

for(let i = 0 ; i<10000; i++){
    arr[selfNum(i)] = true
  }


for(let i = 0; i<10000; i++){
  if(!arr[i]){
    console.log(i)
  }
}

File: legacy-algorithm-study/BOJ/[4673] 셀프넘버.py
def selfNum(a):
    return a+sum([int(i) for i in str(a)])

arr =[0] *10000

for i in range(10000):
    if selfNum(i)<10000:
        arr[selfNum(i)] = 1
        
for j in range(10000):
    if arr[j] == 0:
        print(j)
